fe/src/main/java/org/apache/doris/analysis/DateLiteral.java

```diff
@@ -320,6 +320,13 @@ protected void toThrift(TExprNode msg) {
     @Override
     protected Expr uncheckedCastTo(Type targetType) throws AnalysisException {
         if (targetType.isDateType()) {
+            if (targetType.equals(Type.DATE)) {
+                this.castToDate();                            
+            } else if (targetType.equals(Type.DATETIME)) {
+                this.type = Type.DATETIME;                            
```

Comment:
If Date not to convert DateTime, is OK?

Reply:
Yes, is ok。And explicit type conver(cast expr) also go to this line